### PR TITLE
Restore initial AzContext for New-TestResources

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -125,366 +125,374 @@ $exitActions = @({
...
-    $templateFile = $_.FullName
-
-    Write-Verbose "Found template '$templateFile'"
-    $templateFiles += $templateFile
-}
-
-if (!$templateFiles) {
-    Write-Warning -Message "No template files found under '$root'"
-    exit
+New-Variable -Name 'initialContext' -Value (Get-AzContext) -Option Constant
```
Comment (pakrym) [inline eng/common/TestResources/New-TestResources.ps1:128]: Why not $initialContext = Get-AzContext?

**Comment by pakrym** [comment]
Can confirm. This helps with ctrl+c issue I've been having.

File: eng/common/TestResources/New-TestResources.ps1 (hunk 1 above)
Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:128]: Extra protection to make sure it's not overwritten accidentally (see `-Option Constant`).

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:128]: Why is this particular variable warrant the extra protection and not others?

File: eng/common/TestResources/New-TestResources.ps1 [hunk 2]
```diff
@@ -125,374 +125,387 @@ $exitActions = @({
...
-# If no base name is specified use current user name
-if (!$BaseName) {
-    $BaseName = "$UserName$ServiceDirectory"
+    $UserName =  if ($env:USER) { $env:USER } else { "${env:USERNAME}" }
+    # Remove spaces, etc. that may be in $UserName
+    $UserName = $UserName -replace '\W'
+
+    # Make sure $BaseName is set.
+    if ($CI) {
+        $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
```
Comment (pakrym) [inline eng/common/TestResources/New-TestResources.ps1:166]: Are we always overriding the parameter here?

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:166]: For CIs, yes. It's been this way for a long time, but was lower down. Figured I'd collocate them for easier maintenance.

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:166]: Daniel and I used to calculate them outside and pass it in, but even back then we realized putting more into the script was easier than maintaining separate scripts (inline or not) in the pipelines.

Comment (pakrym) [reply eng/common/TestResources/New-TestResources.ps1:166]: It's a little bit strange. Should we stop passing it all together and only override the value when not provided?

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:166]: No. I do use it for some tests, namely Search because for Search + Key Vault testing it requires that a Key Vault has purge protection, which means I can't purge it and need to use a different name for subsequent testing until it's automatically purged (minimum 7 days, which is what I updated the template to use).

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:166]: Also, for those who want to use this for more long-term resource provisioning (I did that for quite a time and know others who do), I'm sure no one wants to have to remember which t<random digits> resource group is theirs. Maybe we could always tag resources, but it seems like trying to make something that could be easy (username as part of resource names) harder for no other reason than not have a condition in a script (that already has lots of other conditions). I don't see the value in changing this.

**Review by pakrym — APPROVED** [review]

**Review by benbp — APPROVED** [review]
